qiskit/transpiler/passes/optimization/swap_cancellation.py

```diff
@@ -0,0 +1,54 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Cancel back-to-back `swap` gates in dag."""
+
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class SWAPCancellation(TransformationPass):
+    """Cancel back-to-back `swap` gates in dag."""
+
+    def run(self, dag):
+        """Run the SWAPCancellation pass on `dag`.
+
+        Args:
+            dag (DAGCircuit): the directed acyclic graph to run on.
+
+        Returns:
+            DAGCircuit: Transformed DAG.
+        """
+        swap_runs = dag.collect_runs(["swap"])
```

Comment:
Does this actually work as expected? The `collect_runs` method was mostly for 1q gates I don't think we have any tests of using it with >1q gates. Looking at the rustworkx code for the function (https://github.com/Qiskit/rustworkx/blob/main/src/dag_algo/mod.rs#L460-L506 ) it should work I think, but having a testing with partially overlapping swaps (eg, swap(0,1) -> swap(1, 2) -> swap (2, 0)) to make sure the behavior is sound